feat: single commit view header

cloudhead opened the pull request:
This only includes the commit header. No changeset yet, since this needs to be provided by coco. Note that since we don't have the *user* from a commit, just author/email, I've deviated slightly from the original [design](https://www.figma.com/file/dnYlebm0ehgyJyT3Po9Kcl/Upstream?node-id=5995%3A2425).

![2020-04-09-152625_1226x685_scrot](https://user-images.githubusercontent.com/40774/78899781-2abd4c80-7a76-11ea-940b-bfd355f08be9.png)

Addresses parts of #274

cloudhead commented on ui/DesignSystem/Primitive/Title.svelte:
```diff
@@ -17,6 +17,12 @@
+    font-family: var(--typeface-bold);
+    font-size: 20px;
+    line-height: 24px;
+  }
```
This size was missing.

cloudhead commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+      const commit = result.data.commit;
+
+      // TODO(cloudhead): Fetch branch from backend.
+      commit.branch = "master";
```
We need to provide this somehow from the proxy, when fetching a commit.

xla replied:
What are the semantics of branch association for commits? How do we deal with it occurring in multiple? Is this specced already? If so we can then evaluate if we have the building blocks to implement it or need more functionality in surf.

xla replied:
Would also reckon that in this view it's important to not only show the branch but the `repo`, `branch` tuple unless we solve that in another way.

cloudhead commented:
Early review on this would be nice, and also discussion on how/if to include branch name in commit payload.

xla commented on ui/DesignSystem/Component/SourceBrowser/CommitTeaser.svelte:
```diff
@@ -1,7 +1,11 @@
   import UserCard from "../UserCard.svelte";
+  import * as path from "../../../lib/path.js";
+
+  const projectId = getContext("projectId");
```
Why did you decide to pass this through the context instead of a component prop? Where is the context set?

rudolfs replied:
I agree, we should pass it in via a prop and the callers should be adjusted accordingly.

xla commented on ui/DesignSystem/Component/SourceBrowser/CommitTeaser.svelte:
```diff
@@ -1,7 +1,11 @@
 <script>
```
Is this component used anywhere?

cloudhead replied:
Yeah, it's what is shown above source files for eg.

xla commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+  const projectId = params.id;
+  const commitHash = params.hash;
+
+  const QUERY = gql`
```
```suggestion
  const GET_COMMIT = gql`
```

cloudhead replied:
Done. d85aabf

xla commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+  async function fetchCommit() {
+    try {
+      const response = await query(getClient(), {
+        query: QUERY,
```
```suggestion
        query: GET_COMMIT,
```

cloudhead commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+        {commit.summary}
+      </Title>
+      <span class="field">
+        Committed to
```
That's what I thought, but unfortunately `Text` is a `<p>`, which doesn't work here. I discussed this with rudolfs some time ago, I don't think we should have a `Text` component, but rather unpack it into `<Body>`, `<Label>` etc.

cloudhead replied:
Actually it might work for this specific case, but not the other `<span>`..

cloudhead commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+          Commit
+          <a href={path.projectCommit(projectId, commit.sha1)} use:link>
+            {commit.sha1}
+          </a>
```
Yeah it does, but commit hashes are always links as far as I can tell.

cloudhead replied:
Ok checked with Julien, we're going to make this not a link.

cloudhead commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+    <!-- TODO(cloudhead): Should be <pre> -->
+    <Code variant="code" style="margin-bottom: 1rem">{commit.message}</Code>
+    <hr />
+    <Flex align="space-between">
```
Ah yes you might be right, I forgot about the slot thing.

cloudhead replied:
Ok I'm not sure this will work, the complication is that, if you look at the design file, we may sometimes get merge commit information, as well as commiter/author distinction, and want to bottom align in case there are two rows on one side and one row on the other side.

cloudhead replied:
Nevermind, got it working!

cloudhead replied:
083f443

MeBrei replied:
adding `style="align-items: flex-end;"` to the `Flex` in my code snippet, solves that problem.

cloudhead commented on proxy/src/graphql/schema.rs:
```diff
@@ -510,6 +510,10 @@ impl coco::Commit {
+
     fn committer_time(&self) -> String {
         self.committer_time.seconds().to_string()
     }
```
Should I remove this now? Only problem is that the time won't be converted to seconds anymore.

xla replied:
I think you can't remove this as juniper won't know how to convert that type.

cloudhead commented on ui/DesignSystem/Primitive/Icon/Branch.svelte:
```diff
@@ -1,13 +1,14 @@
 <script>
   export let style = null;
+  export let color = "foreground-level-5";
```
Oh cool. 9a47b9b

cloudhead commented on ui/Screen/Project/Commit.svelte:
```diff
@@ -0,0 +1,131 @@
+  import { showNotification } from "../../store/notification.js";
+  import { getClient, query } from "svelte-apollo";
+  import { format } from "timeago.js";
+  import * as path from "../../lib/path.js";
```
f002076

cloudhead commented:
I've addressed review feedback, thanks!

Since most of the missing functionality requires some changes downstream from `surf`, I'd like to get this merged with the current set of functionality, to not keep things lingering too long. I'm tracking the missing pieces here: https://github.com/radicle-dev/radicle-upstream/issues/274

Another big piece that is missing, that @xla brought up is everything around the "repository". Since that is a cross-cutting change, I also want to address it separately.
